src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,162 @@
 #
 #
 
-.rs.addJsonRpcHandler("preview_data_import", function(path, maxRows, maxCols = 100, maxFactors = 64)
+.rs.addFunction("assemble_data_import_name", function(dataImportOptions)
+{
+  dataName <- dataImportOptions$dataName
+
+  if (is.null(dataName) || dataName == "")
+    {
+      if (!is.null(dataImportOptions$importLocation) && dataImportOptions$importLocation != "")
+      {
+        basename <- basename(dataImportOptions$importLocation)
+        if (length(basename) > 0)
+        {
+          fileComponents <- unlist(strsplit(basename, ".", fixed = TRUE))
+          components <- length(fileComponents)
+          if (components >= 1)
+          {
+            dataName <- paste(fileComponents[1:components-1])
+          }
+        }
+      }
+    }
+  
+  dataName
+})
+
+.rs.addFunction("assemble_data_import_parameters_csv", function(dataImportOptions)
+{
+  parameters <- ""
+  spacing <- ",\n  "
+  useCsv <- (!is.null(dataImportOptions$delimiter) && dataImportOptions$delimiter == ",")
+
+  if (!is.null(dataImportOptions$delimiter) && dataImportOptions$delimiter != ",")
+  {
+    parameters <- paste0(parameters, spacing, "delim = \"", dataImportOptions$delimiter, "\"")
```

Comment:
Note that we can't generally use `paste0` since we need to be compatible with old versions of R (`paste0` didn't exist until R 3.0). You can use `paste(..., sep = "")` for the same effect. 

In this case however we're already depending on `readr`, which depends on R >= 3.0.2 (you can find this information in a package's DESCRIPTION file), so it's fine as long as this code is unreachable from older R.